File: Electronics_Shop_17.Model/Requests/AddCustomer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Electronics_Shop_17.Model.Requests
{
    public class AddCustomer
    {
        public int LoyaltyPoints { get; set; } = 0;

        public int UserAccountId { get; set; }
        public virtual List<AddAdress> Adresses { get; set; }
        public virtual List<AddPaymentMethod> PaymentMethods { get; set; }
        public int PersonId { get; set; }
    }
}

File: angular17_web/src/app/models/customer/add.customer.ts
import { AddAdress } from '../adress/add.adress';
import { AddPaymentMethod } from '../payment-method/add.payment.method';
import { AddPerson } from '../person/add.person';

export class AddCustomer {
  loyaltyPoints?: number;
  userAccountId!: number;
  adresses!: AddAdress[];
  paymentMethods!: AddPaymentMethod[];
  personId!: number;
  person!: AddPerson;
  isDeleted?: boolean;

  constructor(data?: any) {
    if (!data) return;

    this.loyaltyPoints = data.loyaltyPoints;
    this.userAccountId = data.userAccountId;
    this.adresses = Array.isArray(data.adresses)
      ? data.adresses.map((x: any) => new AddAdress(x))
      : [];
    this.paymentMethods = Array.isArray(data.paymentMethods)
      ? data.paymentMethods.map((x: any) => new AddPaymentMethod(x))
      : [];
    this.personId = data.personId;
    this.person = data.person ? new AddPerson(data.person) : undefined as any;
    this.isDeleted = data.isDeleted;
  }
}
